monitor.cpp

```diff
@@ -3,36 +3,42 @@
 #include <thread>
 #include <chrono>
 #include <iostream>
+#include <string>
+
 using std::cout, std::flush, std::this_thread::sleep_for, std::chrono::seconds;
 
-int vitalsOk(float temperature, float pulseRate, float spo2) {
-  if (temperature > 102 || temperature < 95) {
-    cout << "Temperature is critical!\n";
-    for (int i = 0; i < 6; i++) {
-      cout << "\r* " << flush;
-      sleep_for(seconds(1));
-      cout << "\r *" << flush;
-      sleep_for(seconds(1));
-    }
-    return 0;
-  } else if (pulseRate < 60 || pulseRate > 100) {
-    cout << "Pulse Rate is out of range!\n";
-    for (int i = 0; i < 6; i++) {
-      cout << "\r* " << flush;
-      sleep_for(seconds(1));
-      cout << "\r *" << flush;
-      sleep_for(seconds(1));
+constexpr float MAX_TEMP = 102;
+constexpr float MIN_TEMP = 95;
+constexpr float MAX_PULSE = 100;
+constexpr float MIN_PULSE = 60;
+constexpr float MIN_SPO2 = 90;
+constexpr float MAX_SPO2 = 100;
+const char* tempWarning = "Temperature is critical!\n";
+const char* pulseWarning = "Pulse Rate is out of range!\n";
+const char* oxygenWarning = "Oxygen Saturation out of range!\n";
+
+bool isVitalNormal(float value, float max, float min, const char* vitalWarning) {
```

Comment:
Nice isolation of range-comparison. Notice, that by doing this, 100% coverage is got without additional tests :)